java/src/main/java/com/microsoft/onnxruntime/ONNXUtil.java

```diff
@@ -0,0 +1,648 @@
+/*
+ * Copyright © 2019, Oracle and/or its affiliates. All rights reserved.
+ */
+package com.microsoft.onnxruntime;
+
+import java.util.Arrays;
+
+import static com.microsoft.onnxruntime.TensorInfo.MAX_DIMENSIONS;
+
+/**
+ * Util code for interacting with shape arrays.
+ */
+public class ONNXUtil {
+    /**
+     * Creates a new primitive boolean array of up to 8 dimensions, using the supplied shape.
+     * <p>
+     * Does not check the shape to see if all it's elements are positive.
+     *
+     * @param shape The shape of array to create.
+     * @return A boolean array.
+     */
+    public static Object newBooleanArray(long[] shape) {
+        switch (shape.length) {
```

Comment:
Can we represent tensors as a contiguous array with the shape stored inside it?

Reply:
This is for interop with regular Java arrays. I could make the return value for some classes (ONNXTensor principally) into a subclass of `Buffer` (e.g. FloatBuffer), and make a tuple with a shape. There isn't a Java tensor type in the JDK yet, and so I think there still needs to be an interop path to a multidimensional array, even if it's slow.

Reply:
As an example C# has a [System.Numerics.Tensor type](https://github.com/dotnet/corefx/tree/master/src/System.Numerics.Tensors/src/System/Numerics/Tensors) that we use to represent tensors in our C# API. It allows the implementation of different types of Tensors (sparse tensors for e.g.). DenseTensor stores values in a contiguous sequential block of memory. I understand that we don't have a first class Tensor type in JDK yet. There's no harm defining one and including it as part of this project to start with. What do you think?

Reply:
I fully support single continuous buffer. It would be even better if somehow we could expose an interface that works with the native buffer.

Reply:
Well, I'm also working in the Tensorflow JVM SIG which is defining a general Tensor type, and I'm investigating writing one on top of the native memory support coming in the OpenJDK Panama project. I'm rather wary of having each library pick it's own Tensor class, with no interop between them. And for the time being they will need to interop with each other, and the only solutions there are `ByteBuffer` or multidimensional arrays.

Reply:
Sorry, missed a comment there. I can expose per element accessors on ONNXTensor, but that will require the user to use the appropriately typed one for their Tensors. Or have it return boxed types which will be detrimental to performance. For some of the primitive types I can wrap the underlying OrtTensor data pointer as a ByteBuffer and then use that to avoid paying the JNI cost every time, but that's going to involve replicating the indexing logic in Java.

Reply:
Wrapping the underlying OrtTensor data pointer as a ByteBuffer sounds like a good idea. It allows us to change the implementation of OrtTensor later on while keeping the interface intact.

Reply:
Yes, that's the right idea. `ByteBuffer` is limited to 32-bit indexes and won't work with buffers larger than 2 GB, so it shouldn't be exposed to users.